lib/Config.js

```diff
@@ -389,6 +397,15 @@ class Config extends EventEmitter {
                     });
                 }
             });
+            const defaultEndpoint = replicationEndpoints.find(
+                endpoint => endpoint.default);
+            if (Object.keys(this.locationConstraints)
+                .map(k => this.locationConstraints[k])
+                .some(location => location.isTransient)) {
+                assert(defaultEndpoint.type, 'bad config: ' +
```

Comment:
I think it's worth noting if there is only one replication endpoint the default can be omitted. See: https://github.com/scality/S3/pull/1184/files#diff-9280903abc69f09eff9ab36070b6d751R355. I don't think it would make sense to have a transient location with only one replication endpoint, but just want to make sure we don't happen to run into a problem accessing `type` of `undefined`.

Reply:
ok will add a check for this, thanks!